src/dom/dom.js

```diff
@@ -412,16 +412,21 @@ p5.prototype.createImg = function() {
   var elt = document.createElement('img');
   var args = arguments;
   var self;
-  elt.alt = args[1];
+  if (args.length > 1 && typeof args[1] === 'string') {
+    elt.alt = args[1];
+  }
   if (args.length > 2 && typeof args[2] === 'string') {
     elt.crossOrigin = args[2];
   }
-  elt.addEventListener('load', function() {
-    self.width = elt.offsetWidth || elt.width;
-    self.height = elt.offsetHeight || elt.height;
-    var last = args[args.length - 1];
-    if (typeof last === 'function') last();
-  });
+  elt.addEventListener(
+    'load',
+    function() {
+      self.width = elt.offsetWidth || elt.width;
+      self.height = elt.offsetHeight || elt.height;
+      var last = args[args.length - 1];
+      if (typeof last === 'function') last((self = addElement(elt, this)));
```

Comment:
Having `addElement(elt, this)` here as well as on the last line of the `createImg` function means we'll actually create two elements with this image source. i think we could instead set `self = addElement(elt, this)` once, right before we create this event listener. Then could pass `self` to the callback function, as well as `return` it. What do you think?

Reply:
Totally agreed on that!